### DISCUSSION TITLE
Problem while traininig the multi speaker VITS on custom dataset

### DISCUSSION THREAD

**Discussion by alifarrokh** [coqui-ai/TTS#2011] [General Q&A]
HI. I'm trying to train the multi speaker VITS model on a custom persian dataset. The dataset is in `lj_speech` format and the `metadata.txt` file **already contains the phoneme sequence** for each audio file.

Here's a few samples from `metadata.txt`:
```txt
31-2-107.wav|piSravi dar zamAn be meqdAre moSaxxas be yek lahzeye moSaxxas *eqdAmAte *imeni
31-2-50.wav|bAre digar Soru* kardam be bastanaS *ammA yek lahze sabr kon bebinam
31-2-115.wav|bAzgaSt *istgAhhAye motevaqqefsAziye zamAn *istgAhhAye parvAz dar zamAn jobrAne *ertefA* va
```

I'm using [this recipe](https://github.com/coqui-ai/TTS/blob/dev/recipes/vctk/vits/train_vits.py) to train the model. Here's the config:
```python
config = VitsConfig(
    model_args=vitsArgs,
    audio=audio_config,
    run_name="vits_vctk",
    batch_size=8,
    eval_batch_size=8,
    batch_group_size=5,
    num_loader_workers=1,
    num_eval_loader_workers=1,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=200,
    text_cleaner="identity_cleaner",
    use_phonemes=False,
    # phoneme_language="fa",
    # phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    # compute_input_seq_cache=True,
    print_step=24,
    save_step=24,
    print_eval=False,
    mixed_precision=True,
    max_text_len=325,  # change this if you have a larger VRAM than 16GB
    output_path=output_path,
    datasets=[dataset_config],
    cudnn_benchmark=False,
    characters=CharactersConfig(
        characters_class = "TTS.tts.models.vits.VitsCharacters",
        pad = "_",
        eos = "~",
        bos = "",
        characters = char_set,
        punctuations = " ",
        # phonemes = char_set,
        is_unique = True,
        is_sorted = False
    ),
    test_sentences=[],
)
```
- `identity_cleaner` is a cleaner function defined in `cleaners.py`. It simply returns the text itself since the phoneme sequence is already cleaned.
- `char_set` is a string containing the list of all unique characters used in the phoneme sequence.

Firstly, I'm training the model on a small 50-sample dataset in order to check if the model can overfit the data and generate the same samples in the output. However, after 200 epochs, the synthesized samples are nothing but noise. Any idea on how to find the problem?

Thanks in advance.

**Comment by alifarrokh** [comment]
I also tried to train the model on VCTK dataset with the same config, yet not working.

**Reply by WeberJulian** [reply]
Are you trying to add Persian to all the languages of MAILABS  or you want a Persian only model?

**Reply by alifarrokh** [reply]
@WeberJulian I want a Persian-only model.

**Reply by WeberJulian** [reply]
Sorry, I read multilingual not multispeaker. Also why train multispeaker? Your `metadata.txt` file looks single speaker. Have you removed this code from the recipe?

```
speaker_manager = SpeakerManager()
speaker_manager.set_ids_from_data(train_samples + eval_samples, parse_key="speaker_name")
config.model_args.num_speakers = speaker_manager.num_speakers 
```

Maybe try from the ljspeech vits recipe instead.
Also maybe 50 samples for the dataset caused issue for splitting eval and dev. Verify that you have at least one batch of train data.

**Reply by pooriaarab** [reply]
@alifarrokh heyy i'm trying to clone a persian voice.

got any tips or guide on how to do it using this repo?
